NBitcoin/IpExtensions.cs

```diff
@@ -138,6 +138,36 @@ public static bool IsRFC4843(this IPAddress address)
 			return (bytes[15 - 15] == 0x20 && bytes[15 - 14] == 0x01 && bytes[15 - 13] == 0x00 && (bytes[15 - 12] & 0xF0) == 0x10);
 		}
 
+		public static byte[] GetGroup(this EndPoint endpoint)
+		{
+			if (endpoint is IPEndPoint ip)
+			{
+				return ip.Address.GetGroup();
+			}
+			else if (endpoint is DnsEndPoint dns)
+			{
+				var vchRet = new List<byte>();
+				if (dns.IsTor() || dns.IsI2P())

```

Comment:
Are you sure you wanted to do this? Should we throw error if it's a `DnsEndPoint` but not Tor, nor I2P? It looks to me you're building the group without this condition just fine, too.

Reply:
IDK. The logic behing grouping addresses is that an attacker would need to get addresses in many groups in order to isolate/eclipse a node however those addresses that are not Tor v3 or I2P can be whatever or in most of the cases they would be simply host names translatable to IP addresses so they should belong to the same group that theirs corresponding IPs but that would require to query a DNS and even in that case a host can have multiple ip addresses so, it is hard to say. I preferred this to close the door to any unexpected situation.

Reply:
ACK